content/docs/user-guide/experiment-management/cleaning.md

```diff
@@ -0,0 +1,262 @@
+
+Issuing `dvc exp gc --workspace` removes experiments in `other` and `another`
+branches in this example.
+
+```dvc
+$ dvc exp gc --workspace
+$ dvc exp list --all
+main:
+   exp-abc000
+   exp-abc111
+   exp-abc222
+```
+
+### Keeping experiments in all branches
+
+DVC can create a branch for an experiment using `dvc exp branch` command.
+
+In cases where you want to clean up the experiments _except_ those in the
+branches, you can use `--all-branches` flag.
+
+```dvc
+$ dvc exp show --all-branches
+```
+
+```dvctable
+┏━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━┳━━━━━━━━┳━━━━━━━━━━━━━━━━━━┓
+┃ Experiment              ┃ Created      ┃    acc ┃ model.conv_units ┃
+┡━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━╇━━━━━━━━╇━━━━━━━━━━━━━━━━━━┩
+│ workspace               │ -            │      - │ 64               │
+│ cnn-48                  │ 09:11 AM     │ 0.9131 │ 48               │
+│ main                    │ Jul 21, 2021 │ 0.9189 │ 16               │
+│ ├── dac711b [cnn-32]    │ 09:16 AM     │ 0.9152 │ 32               │
+│ ├── 7cd3ae7 [cnn-48]    │ 09:11 AM     │ 0.9131 │ 48               │
+│ ├── ab585b5 [cnn-24]    │ 09:06 AM     │ 0.9135 │ 24               │
+│ ├── 7d51b55 [exp-44136] │ 09:01 AM     │ 0.9151 │ 16               │
+│ └── 7feaa1c [exp-78ede] │ Aug 02, 2021 │ 0.9151 │ 16               │
+│ 8583124                 │ Jul 20, 2021 │ 0.9132 │ 17               │
+└─────────────────────────┴──────────────┴────────┴──────────────────┘
+```
+
+Supplying `--all-branches` keeps only the experiments in branch tips. Any
+experiment that's not promoted to a branch is removed this way.
+
+```dvc
+$ dvc exp gc --all-branches
+WARNING: This will remove all experiments except those derived from the workspace and all git branches of the current repo.
+Are you sure you want to proceed? [y/n] y
+Removed 6 experiments. To remove unused cache files use 'dvc gc'.
+```
+
+The resulting `dvc exp show` table is as the following:
+
+```dvctable
+┏━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━┳━━━━━━━━┳━━━━━━━━━━━━━━━━━━┓
+┃ Experiment              ┃ Created      ┃    acc ┃ model.conv_units ┃
+┡━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━╇━━━━━━━━╇━━━━━━━━━━━━━━━━━━┩
+│ workspace               │ -            │      - │ 64               │
+│ cnn-48                  │ 09:11 AM     │ 0.9131 │ 48               │
+│ main                    │ Jul 21, 2021 │ 0.9189 │ 16               │
+│ 8583124                 │ Jul 20, 2021 │ 0.9132 │ 17               │
+└─────────────────────────┴──────────────┴────────┴──────────────────┘
+```
+
+### Keeping experiments in all tags
+
+- [ ] `--all-tags` example
+
+### Keeping experiments in all commits
+
+- [ ] `--all-commits` example
+
+## Removing Experiments in Remotes
+
+As you push the experiments with `dvc exp push`, remotes may be become cluttered
+with experiment references.
+
+DVC doesn't provide a shortcut for cleaning up the experiments in remotes but
+you can use Git plumbing commands to remove experiment references from remotes.
```

Comment:
There's no `exp gc -c`? Hmmm sounds like a missing feature. Probably better to open a feature request than document the hole.

Reply:
We communicated with @pmrowla about this in Slack, and this is actually his explanation. There is already an issue in iterative/dvc#6006

Reply:
Maybe we should prioritize this then? cc @dberenbaum

`ls-remote` is [a plumbing command](https://git-scm.com/docs/git#Documentation/git.txt-ahrefdocsgit-ls-remotegit-ls-remote1a). Also makes the user care about `refs`, which they normally should not? 🙂

Reply:
With my communication with @pmrowla, I got the impression that it will take a while to implement this. I believe this is a common use case, at least I needed for myself and cleaning up a common experiments server is not that rare. We can update the docs when the feature arrives.

Reply:
So do we want to keep this information for now or nay?

Reply:
p.s. I see iterative/dvc/issues/6006 was assigned recently.

Reply:
I think I still need some days to implement this, so you can keep this info for now.

Reply:
Sorry, As far as I know, we don't have any plan for this in the near future. Most of our attention is on the remote execution side.

Reply:
There is a plan for removing experiments from remotes: https://github.com/iterative/dvc/pull/6471. I think this should be enough to address the note here?